plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php

```diff
@@ -117,14 +118,45 @@ public function is_complete() {
 			$third_party_complete = apply_filters( 'woocommerce_admin_third_party_tax_setup_complete', false );
 
 			$this->is_complete_result = $is_wc_connect_taxes_enabled ||
-				count( TaxDataStore::get_taxes( array() ) ) > 0 ||
-				get_option( 'woocommerce_no_sales_tax' ) !== false ||
-				$third_party_complete;
+				false !== get_option( 'woocommerce_no_sales_tax' ) ||
+				$third_party_complete ||
+				$this->is_actioned();
+
+			if ( ! $this->is_complete_result && $this->has_existing_tax_rates() ) {
+				$this->mark_actioned();
```

Comment:
We should also add `$this->is_complete_result = true;` in here otherwise this function still returns `false` the first time even though its actioned.  This results in the task being marked incomplete the first time and complete on subsequent reloads.

**First load**

<img width="720" alt="Screenshot 2025-03-17 at 3 32 04 PM" src="https://github.com/user-attachments/assets/f5592c17-b70a-40b0-99e0-70a4b491be3f" />

**Reload**
<img width="716" alt="Screenshot 2025-03-17 at 3 33 06 PM" src="https://github.com/user-attachments/assets/ae3e28da-8e35-441b-a6b6-cc51611d8747" />